### ISSUE TITLE
4.4 silently changes the Inspect API

### ISSUE THREAD

**Issue by ShaheedHaque** [celery/celery#5890]
# Checklist

- [x] I have checked the [issues list](https://github.com/celery/celery/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22Category%3A+Documentation%22+)
  for similar or identical bug reports.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the bug was already fixed in the master branch.
- [x] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Description

Prior to Celery 4.4.0, the celery.app.control.Inspect.scheduled() API used to return "args" and "kwargs" as strings which could be parsed using ast.literal_eval. Now, they seem to be a list and dict respectively:

 {'id': 'd82058c7-12b0-4914-8381-2fef9ad874a9', 'name': '...', 'args': ['time', 'sleep', 5], 'kwargs': {}, 'type': '...', }

# Suggestions

I assume this is a deliberate design change; it would have been helpful to add this breaking change to the release notes!

**Comment by auvipy** [comment]
I admit the release notes are not complete as I released the 4.4.0 quite hastily before the holiday season starts. thanks for the report

**Comment by kitsuyui** [comment]
I think this issue is related to: https://groups.google.com/g/celery-users/c/1F61iEj-YFI/m/N7d64AalDAAJ

https://github.com/celery/celery/blob/8492b75c579564c2af5c2be75fe4b2118ebd0cd1/celery/worker/request.py#L558-L559

args and kwargs are passed directly. So correctly working when using JSON serialization.
But when using pickle serialization, it causes error.

```
kombu.exceptions.EncodeError: Object of type **** is not JSON serializable
```

## Reproduction

`celery -A repro worker` and `python repro.py` in separated two shells.

```python
import time

from celery import Celery


app = Celery('example')
app.conf.update(
    backend_url='redis://localhost:6379',
    broker_url='redis://localhost:6379',
    result_backend='redis://localhost:6379',
    task_serializer='pickle',
    accept_content=['pickle', 'json'],
)


@app.task(name='task1')
def task1(*args, **kwargs):
    print('Start', args, kwargs)
    time.sleep(30)
    print('Finish', args, kwargs)


def main():
    task1.delay({1, 2, 3})  # something not JSON serializable.

    # inspect queue items
    inspected = app.control.inspect()
    active_tasks = inspected.active()
    print(active_tasks)


if __name__ == '__main__':
    main()
```

**Comment by auvipy** [comment]
> I think this issue is related to: https://groups.google.com/g/celery-users/c/1F61iEj-YFI/m/N7d64AalDAAJ
> 
> https://github.com/celery/celery/blob/8492b75c579564c2af5c2be75fe4b2118ebd0cd1/celery/worker/request.py#L558-L559
> 
> args and kwargs are passed directly. So correctly working when using JSON serialization.
> But when using pickle serialization, it causes error.
> 
> ```
> kombu.exceptions.EncodeError: Object of type **** is not JSON serializable
> ```
> 
> ## Reproduction
> 
> `celery -A repro worker` and `python repro.py` in separated two shells.
> 
> ```python
> import time
> 
> from celery import Celery
> 
> 
> app = Celery('example')
> app.conf.update(
>     backend_url='redis://localhost:6379',
>     broker_url='redis://localhost:6379',
>     result_backend='redis://localhost:6379',
>     task_serializer='pickle',
>     accept_content=['pickle', 'json'],
> )
> 
> 
> @app.task(name='task1')
> def task1(*args, **kwargs):
>     print('Start', args, kwargs)
>     time.sleep(30)
>     print('Finish', args, kwargs)
> 
> 
> def main():
>     task1.delay({1, 2, 3})  # something not JSON serializable.
> 
>     # inspect queue items
>     inspected = app.control.inspect()
>     active_tasks = inspected.active()
>     print(active_tasks)
> 
> 
> if __name__ == '__main__':
>     main()
> ```

would you mind finding a possible fix?

**Comment by kitsuyui** [comment]
I think there is a possible fix like this https://github.com/celery/celery/pull/6567

**Comment by matusvalo** [comment]
We need to understand how we should fix this issue either via PR #6567 or via my comment: https://github.com/celery/celery/pull/6567#issuecomment-788008337
